EventListener/Doctrine/CleanListener.php

```diff
@@ -2,43 +2,47 @@
 
 namespace Vich\UploaderBundle\EventListener\Doctrine;
 
-use Doctrine\Common\EventArgs;
+use Doctrine\ORM\Event\PreFlushEventArgs;
+use Doctrine\ORM\Events;
 
 /**
  * CleanListener
  *
  * Listen to the update event to delete old files accordingly.
  *
+ * @author Konstantin Myakshin <koc-dp@yandex.ru>
  * @author Kévin Gomez <contact@kevingomez.fr>
  */
 class CleanListener extends BaseListener
 {
     /**
-     * The events the listener is subscribed to.
-     *
-     * @return array The array of events.
+     * {@inheritdoc}
      */
     public function getSubscribedEvents()
     {
-        return array(
-            'preUpdate',
-        );
+        return array(Events::preFlush);
     }
 
     /**
-     * @param EventArgs $event The event.
+     * @param PreFlushEventArgs $event
      */
-    public function preUpdate(EventArgs $event)
+    public function preFlush(PreFlushEventArgs $event)
     {
-        $object = $this->adapter->getObjectFromArgs($event);
+        $em = $event->getEntityManager();
+        $identityMap = $em->getUnitOfWork()->getIdentityMap();
```

Comment:
How can I got changed entities in other way? `PreFlush` event not recieve entities